### PR TITLE
Support showing knowledge base

### PR DESCRIPTION
**Pull request by jaypume**
1. Support showing knowledge base in crd status field.
2. Add knowledge base types definition.
3. Add autogen crd related code based on types.go.
4. Add some comment.

### CODE REVIEW

File: pkg/localcontroller/managers/lifelonglearning/lifelonglearningjob.go [hunk 1]
```diff
@@ -281,6 +281,10 @@ func (lm *Manager) evalTask(job *Job) error {
 				jobConfig.UniqueIdentifier, err)
 		}
 
+		// 这个触发器是指，LC已经告诉GM可以触发Eval worker了，
```
Comment (Poorunga) [inline pkg/localcontroller/managers/lifelonglearning/lifelonglearningjob.go:284]: please comment by english

File: pkg/globalmanager/controllers/lifelonglearning/upstream.go [hunk 2]
```diff
@@ -98,6 +99,69 @@ func (c *Controller) appendStatusCondition(name, namespace string, cond sednav1.
...
+		// check if models field exits
+		if cd.Output == nil || cd.Output.Models == nil || len(cd.Output.Models) == 0 {
+			klog.V(4).Infof("output models is nil, name is %s", name)
+			return nil
+		}
+
+		numberOfSamples := 0
+		aiModels := sednav1.AIModels{}
+		aiclasses := sednav1.AIClasses{}
+		aiclasses.ListOfAIClasses = cd.Output.Models[0].Classes
```
Comment (Poorunga) [inline pkg/globalmanager/controllers/lifelonglearning/upstream.go:114]: It is recommended that slice lengths be allocated in advance.

File: pkg/localcontroller/managers/lifelonglearning/lifelonglearningjob.go (hunk 1 above)
Comment (jaypume) [reply pkg/localcontroller/managers/lifelonglearning/lifelonglearningjob.go]: The comment has been changed to English.

File: pkg/globalmanager/controllers/lifelonglearning/upstream.go (hunk 2 above)
Comment (jaypume) [reply pkg/globalmanager/controllers/lifelonglearning/upstream.go]: The following line has beed added to optimize slice allocation. 
`aiModels.ListOfAIModels = make([]sednav1.AIModel, 0, len(cd.Output.Models))`

**Pull request merged**